azurecompute-arm/src/main/java/org/jclouds/azurecompute/arm/compute/functions/VMHardwareToHardware.java

```diff
@@ -48,6 +48,7 @@
    public Hardware apply(VMHardware from) {
       final HardwareBuilder builder = new HardwareBuilder()
               .name(from.name)
+              .providerId("azurecompute-arm")
```

Comment:
This is not the id of the provider. This is the "real" id of the hardware profile in the provider; the "backend id". Some providers such as AWS encode the region information in the IDs, so they have things like "us-east-1/ami-0001", etc. But the +real* id in the provider is "ami-0001".

The `id` field is something free that providers can use to encode the information they need in order to be able to properly implement the "getById" methods in the adapter. The `providerId` is meant to hold the real id of the object in the backend provider.